cirq/experiments/fidelity_estimation.py

```diff
@@ -496,42 +674,51 @@ def sample_2q_xeb_circuits(
             is given by this number.
         progress_bar: A progress context manager following the `tqdm` API or `None` to not report
             progress.
+        combinations_by_layer: Either `None` or the result of
+            `rqcg.get_random_combinations_for_device`. If this is `None`, the circuits specified
+            by `circuits` will be sampled verbatim, resulting in isolated XEB characterization.
+            Otherwise, this contains all the random combinations and metadata required to combine
+            the circuits in `circuits` into wide, parallel-XEB-style circuits for execution.
 
     Returns:
-        A pandas dataframe with index given by ['circuit_i', 'cycle_depth'] and
-        column "sampled_probs".
+        A pandas dataframe with index given by ['circuit_i', 'cycle_depth'].
+        Columns always include "sampled_probs". If `combinations_by_layer` is
+        not `None` and you are doing parallel XEB, additional metadata columns
+        will be attached to the returned DataFrame.
     """
+    # Set up progress reporting
     if progress_bar is None:
         progress_bar = _NoProgress
 
-    q0, q1 = _verify_and_get_two_qubits_from_circuits(circuits)
-    tasks = []
-    for cycle_depth in cycle_depths:
-        for circuit_i, circuit in enumerate(circuits):
-            circuit_depth = cycle_depth * 2 + 1
-            assert circuit_depth <= len(circuit)
-            truncated_circuit = circuit[:circuit_depth]
-            prepared_circuit = truncated_circuit + ops.measure(q0, q1)
-            tasks.append(
-                _Sample2qXEBTask(
-                    cycle_depth=cycle_depth, circuit_i=circuit_i, prepared_circuit=prepared_circuit
-                )
-            )
-
-    n_tasks = len(tasks)
-    batched_tasks = [tasks[i : i + batch_size] for i in range(0, n_tasks, batch_size)]
-
-    run_batch = _SampleInBatches(sampler=sampler, repetitions=repetitions)
-    with ThreadPoolExecutor(max_workers=2) as pool:
-        futures = [pool.submit(run_batch, task_batch) for task_batch in batched_tasks]
-
-        records = []
-        with progress_bar(total=n_tasks) as progress:
-            for future in concurrent.futures.as_completed(futures):
-                records += future.result()
-                progress.update(batch_size)
+    # Shim isolated-XEB as a special case of combination-style parallel XEB.
+    if combinations_by_layer is None:
+        combinations_by_layer, circuits = _get_combinations_by_layer_for_isolated_xeb(circuits)
+        one_pair = True
+    else:
+        _verify_two_line_qubits_from_circuits(circuits)
+        one_pair = False
```

Comment:
instead of one_pair you can create drop_columns = [...]

Reply:
I actually think it reads more intuitively later on when you might be wondering why we're dropping some of the columns: because it's one pair! Also, if there are any other behaviors that need to be slightly modified (other than column dropping) this name would still fit.